test/ServiceTest.cpp

```diff
@@ -187,9 +215,37 @@ TEST_F(ServiceTest, computeResult) {
   Id idBla = Id::makeFromLocalVocabIndex(idxBla.value());
   Id idBlu = Id::makeFromLocalVocabIndex(idxBlu.value());
 
-  // Check that the result table corresponds to the contents of the TSV.
+  // Check that the result table corresponds to the contents of the JSON.
   EXPECT_TRUE(result);
   IdTable expectedIdTable = makeIdTableFromVector(
       {{idX, idY}, {idBla, idBli}, {idBlu, idBla}, {idBli, idBlu}});
   EXPECT_EQ(result->idTable(), expectedIdTable);
 }
+
+// Test that bindingToValueId behaves as expected.
+TEST_F(ServiceTest, bindingToValueId) {
+  Index::Vocab vocabulary;
+  LocalVocab localVocab{};
+
+  nlohmann::json j = {{"type", "literal"}, {"value", "Hello World"}};
+  EXPECT_NO_THROW(Service::bindingToValueId(j, vocabulary, &localVocab));
+
+  j["xml:lang"] = "de";
+  j["value"] = "\"\test\"";
+  EXPECT_NO_THROW(Service::bindingToValueId(j, vocabulary, &localVocab));
+
+  j.erase("xml:lang");
+  j["datatype"] = XSD_INT_TYPE;
+  j["value"] = "42";
+  EXPECT_NO_THROW(Service::bindingToValueId(j, vocabulary, &localVocab));
+
+  j["type"] = "uri";
+  j["value"] = "http://doof.org";
+  EXPECT_NO_THROW(Service::bindingToValueId(j, vocabulary, &localVocab));
+
+  j["type"] = "blank";
+  EXPECT_NO_THROW(Service::bindingToValueId(j, vocabulary, &localVocab));
+
+  j["type"] = "INVALID_TYPE";
+  EXPECT_ANY_THROW(Service::bindingToValueId(j, vocabulary, &localVocab));
```

Comment:
As suggested, if you have an intermediate function that directly returns `TripleComponent`instead of `ID`, this testing becomes much simpler.